File: backend/src/main/java/Dominio/Entity/Entrenador.java
// Dominio/Entity/Entrenador.java
package Dominio.Entity;

import Dominio.Entity.Roles.Roles;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "entrenadores")
@Data
public class Entrenador {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    private String apellido;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(nullable = false)
    private String password;

    @Column(unique = true, nullable = false)
    private String email;

    private int edad;

    @Column(unique = true, nullable = false)
    private String codigoEntrenador;

    @Enumerated(EnumType.STRING)
    private Roles role;

    private boolean verificado;

    private String telefono;

    private String experiencia;

    @OneToOne
    @JoinColumn(name = "usuario_id", unique = true)
    private Usuario usuario;

    @OneToOne(mappedBy = "entrenador")
    private Equipo equipo;

    // Constructor vacío
    public Entrenador() {}

    // Método para obtener nombre completo
    public String getNombreCompleto() {
        return nombre + " " + (apellido != null ? apellido : "");
    }
}

File: frontend/lib/models/entrenador.dart
class Entrenador {
  final int? id;
  final String nombre;
  final String? apellido;
  final String username;
  final String email;
  final int edad;
  final String codigoEntrenador;
  final String? telefono;
  final String? experiencia;
  final bool verificado;
  final bool tieneEquipo;
  final int? equipoId;
  final String? nombreEquipo;

  Entrenador({
    this.id,
    required this.nombre,
    this.apellido,
    required this.username,
    required this.email,
    required this.edad,
    required this.codigoEntrenador,
    this.telefono,
    this.experiencia,
    this.verificado = false,
    this.tieneEquipo = false,
    this.equipoId,
    this.nombreEquipo,
  });

  factory Entrenador.fromJson(Map<String, dynamic> json) {
    int _parseInt(dynamic value) {
      if (value == null) return 0;
      if (value is int) return value;
      if (value is String) return int.tryParse(value) ?? 0;
      return 0;
    }

    int? _parseIntNullable(dynamic value) {
      if (value == null) return null;
      if (value is int) return value;
      if (value is String) return int.tryParse(value);
      return null;
    }

    return Entrenador(
      id: _parseIntNullable(json['id']),
      nombre: json['nombre'] ?? '',
      apellido: json['apellido'],
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      edad: _parseInt(json['edad']),
      codigoEntrenador: json['codigoEntrenador'] ?? '',
      telefono: json['telefono'],
      experiencia: json['experiencia'],
      verificado: json['verificado'] ?? false,
      tieneEquipo: json['tieneEquipo'] ?? false,
      equipoId: _parseIntNullable(json['equipoId']),
      nombreEquipo: json['nombreEquipo'],
    );
  }

  String get nombreCompleto => '$nombre ${apellido ?? ''}'.trim();
}
